File: src/net/merayen/elastic/backend/architectures/local/nodes/data_1/LNode.java
package net.merayen.elastic.backend.architectures.local.nodes.data_1;

import net.merayen.elastic.backend.architectures.local.LocalNode;
import net.merayen.elastic.backend.architectures.local.LocalProcessor;

import java.util.Map;

/**
 * Does absolutely nothing. Just a datacontainer for e.g UI objects that stores their data in a node.
 */
public class LNode extends LocalNode {
	public LNode() {
		super(LProcessor.class);
	}

	@Override
	protected void onInit() {}

	@Override
	protected void onSpawnProcessor(LocalProcessor lp) {}

	@Override
	protected void onProcess(Map<String, Object> data) {}

	@Override
	protected void onParameter(String key, Object value) {}

	@Override
	protected void onFinishFrame() {}

	@Override
	protected void onDestroy() {}
}

File: src/main/kotlin/net/merayen/elastic/backend/architectures/local/nodes/midi_1/LNode.kt
package net.merayen.elastic.backend.architectures.local.nodes.midi_1

import net.merayen.elastic.backend.architectures.local.LocalNode
import net.merayen.elastic.backend.architectures.local.LocalProcessor
import net.merayen.elastic.backend.data.eventdata.MidiData
import net.merayen.elastic.backend.logicnodes.list.midi_1.MidiNodeInputFrameData
import net.merayen.elastic.backend.nodes.BaseNodeData
import net.merayen.elastic.system.intercom.InputFrameData

class LNode : LocalNode(LProcessor::class.java) {
	internal var inputMidi: Array<ShortArray>? = null
	internal var midiData: MidiData? = null
	//internal var bpmCurve: Curv

	override fun onInit() {}

	override fun onSpawnProcessor(lp: LocalProcessor) {}

	override fun onProcess(data: InputFrameData) {
		val input = data as MidiNodeInputFrameData
		if (input.midiDataMessage != null)
			midiData = input.midiDataMessage.midiData

		if (input.temporaryMidi != null)
			inputMidi = input.temporaryMidi
	}

	override fun onParameter(instance: BaseNodeData) {}

	override fun onFinishFrame() {
		inputMidi = null
	}

	override fun onDestroy() {}
}